test/threads.jl

```diff
@@ -0,0 +1,64 @@
+using Test
```

Comment:
We need to also enable `JULIA_NUM_THREADS` in CI.

Reply:
I don't think we have to because this test sets up its own workers with multiple threads in the `exeflags` variable. But I also don't have any objection to starting every worker in the tests with multiple threads anyway, maybe it'd be easier to shake out bugs?

Reply:
> I don't think we have to because this test sets up its own workers with multiple threads in the exeflags variable. 

Ahh yes, I missed that!

> But I also don't have any objection to starting every worker in the tests with multiple threads anyway, maybe it'd be easier to shake out bugs?

It's up to you - I think it could be valuable to try, but maybe we wait for the `@async` -> `Threads.@spawn`  change, to be able to see what happens when tasks get assigned to other threads.

Reply:
Yeah we might as well go for it now, added in 8e35ba4.